sdk-core/src/test/java/care/data4life/sdk/RecordServiceAdditionalResourceTypeTest.kt

```diff
@@ -656,36 +659,44 @@ class RecordServiceAdditionalResourceTypeTest : RecordServiceTestBase() {
                 -1
         )
 
-        val attachments = ArrayList<Attachment>()
-        attachments.add(attachment)
-        attachments.add(secondAttachment)
+        val attachments = ArrayList<WrapperContract.Attachment>()
+        attachments.add(SdkAttachmentFactory.wrap(attachment))
+        attachments.add(SdkAttachmentFactory.wrap(secondAttachment))
+
+        every { 
+            apiService.fetchRecord(ALIAS, USER_ID, RECORD_ID)
+        } returns Single.just(mockEncryptedRecord)
+
+        every {
+            recordServiceK.decryptRecord<Fhir3Resource>(mockEncryptedRecord, USER_ID)
+        } returns decryptedRecord as DecryptedBaseRecord<Fhir3Resource>
+
 
-        Mockito.`when`(mockApiService.fetchRecord(ALIAS, USER_ID, RECORD_ID))
-                .thenReturn(Single.just(mockEncryptedRecord))
-        Mockito.doReturn(decryptedRecord).`when`(recordService)
-                .decryptRecord<DomainResource>(mockEncryptedRecord, USER_ID)
-        Mockito.`when`(mockAttachmentService.download(
-                ArgumentMatchers.argThat { arg -> arg.contains(attachment) },
-                ArgumentMatchers.eq(mockAttachmentKey),
-                ArgumentMatchers.eq(USER_ID))
-        ).thenReturn(Single.just(attachments))
+        every {
+            attachmentService.download(
+                    listOf(
+                            SdkAttachmentFactory.wrap(attachment),
+                            SdkAttachmentFactory.wrap(secondAttachment)
+                    ),
+                    mockAttachmentKey,
+                    USER_ID
+        ) } returns Single.just(attachments)
 
         // when
         val attachmentIds = listOf(ATTACHMENT_ID, secondAttachmentId)
-        val test = recordService.downloadAttachments(RECORD_ID, attachmentIds, USER_ID, DownloadType.Full).test().await()
+        val test = recordServiceK.downloadAttachments(RECORD_ID, attachmentIds, USER_ID, DownloadType.Full).test().await()
```

Comment:
Since the UnitTests are now using both Mockito and mockk, I needed a suffix (or prefix) to differentiate both from each other, and I had no better idea in this moment, so I am open for a better name. However we should somewhen (not now) start to rework the tests, since they are now a awkward hybrid of 2 test frameworks.